src/flow/abstractObjectFactories.js

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+import { Realm } from "../realm.js";
+import buildExpressionTemplate from "../utils/builder.js";
+import { ObjectCreate, ArrayCreate } from "../methods/index.js";
+import { ValuesDomain } from "../domains/index.js";
+import { Value, AbstractValue, ObjectValue, ArrayValue } from "../values/index.js";
+import invariant from "../invariant.js";
+
+type ObjectTypes = {
+  [key: string]: ObjectTypes | string,
+};
+
+export function createObject(realm: Realm, shape: ObjectTypes | null, name: string | null): ObjectValue {
+  let obj = ObjectCreate(realm, realm.intrinsics.ObjectPrototype);
+  if (shape != null) {
+    // to get around Flow complaining that shape could be null
+    let shapeThatIsNotNull = shape;
+    Object.keys(shape).forEach((id: string) => {
+      let value = shapeThatIsNotNull[id];
+      invariant(value instanceof Value, "creation of object failed due to object containing non-value properties");
```

Comment:
ObjectTypes[key] can be `ObjectTypes | string` according to type definition, why will it be a`Value` here?

Reply:
Added `Value` to the union.